### PR TITLE
Markdown / Html Messages include resolved file urls now.

### LINKED ISSUE #29: Use FileLinkSettings to URLs to duplicate fragments
With Cake.PullRequest 0.9.0, we will pass a `FileLinkSettings` to the issue providers (to be implemented), which we can use to resolve links to the exact file and line of an issue. This link can be added to the HTML text to improve the usability of this addin.

### PR DESCRIPTION
**Pull request by janniksam**
**What has changed**

- The provider will provide proper links for html and markdown messages from now on. This needs to be improved in the future, currently the provider has to create additional issues for every fragment, because `GetFileLink` requires an `IIssue` to work. (see [here](https://github.com/janniksam/Cake.Issues.DupFinder/blob/feature/addFileLinkToMessage/src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs#L222))

Closes #29

### CODE REVIEW

File: src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs [hunk 1]
```diff
@@ -79,31 +84,58 @@ private static string GetIdentifier(int cost, IReadOnlyCollection<DuplicateFragm
             return $"{fragment.FilePath}-{cost}-{string.Join("-", otherFragments)}";
         }
 
-        private static string GetHtmlMessage(int cost, IReadOnlyCollection<DuplicateFragment> fragments, DuplicateFragment fragment)
+        private void AddFileLinks(IEnumerable<IIssue> result)
```
Comment (pascalberger) [inline src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs:87]: You don't need to set `IIssue.FileLink) here in the provider as it already will be set by the issue reader

File: src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs [hunk 2]
```diff
@@ -180,7 +212,23 @@ private static bool TryGetLine(XContainer fragment, out int lineStart, out int l
             return true;
         }
 
-        private IReadOnlyCollection<DuplicateFragment> GetDuplicateFragements(XContainer duplicate)
+        private Uri ResolveFileLinkForFragment(DuplicateFragment fragment)
+        {
+            if (string.IsNullOrWhiteSpace(this.Settings?.FileLinkSettings?.FileLinkPattern))
```
Comment (pascalberger) [inline src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs:217]: It should be enough to check that `this.Settings.FileLinkSettings` is not null. There are already checks in place for checking that `this.Settings` is not `null` and `FileLinkPattern` is required to be passed to the ctor of `FileLinkSettings`.

File: src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs (hunk 1 above)
Comment (janniksam) [reply src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs:87]: Where exactly can I see this?

Comment (pascalberger) [reply src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs:87]: Here: https://github.com/cake-contrib/Cake.Issues/blob/release/0.9.0/src/Cake.Issues/IssuesReader.cs#L68

**Pull request merged**